lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -175,3 +175,38 @@ def filter_lines(lines, minimum_length):
         Minimum length for the line to be accepted.
     """
     return [line for line in lines if len(line) >= minimum_length]
+
+
+def refine_lines_centroid(lines, line_width):
+    """Refine the lines based on the brightness-weighted centroid.
+
+    This function uses the pixels in the vicinity of the lines to make small adjustments to the estimated location. The
+    refinement correction is computed by considering the brightness weighted centroid:
+
+      m0 = sum_{i <= w}(I(x + i))
+      dx = (1 / m0) * sum_{i <= w} i(I(x + i))
+
+    Parameters
+    ----------
+    lines : List[pylake.KymoLine]
+        Detected traces on a kymograph.
+    line_width : int
+        Line width
+    """
+    lines = [line.interpolate() for line in lines]
```

Comment:
Given that `lines` is already a parameter, it would be better to have this as `interpolated_lines` or `it_lines` (if you need a shorter name) for readability. 

But now I'm curious: Doesn't this modify the parameter as it is taken by reference?

Reply:
Yeah, that's fair. I'll rename it to `interpolated_lines`.

And no, it shouldn't. If you look at `interpolate`, you can see it explicitly makes a new `KymoLine`. Just to verify:
![image](https://user-images.githubusercontent.com/19836026/100625542-f5c37a00-3324-11eb-96ac-b7bc1e7c468f.png)

Do you want me to add an explicit test for this?

Reply:
>And no, it shouldn't. If you look at interpolate, you can see it explicitly makes a new KymoLine. Just to verify:

Oh, but I meant assigning the list comprehension to `lines`.

>Do you want me to add an explicit test for this?

I don't believe this is needed.

Reply:
> Oh, but I meant assigning the list comprehension to `lines`.

Ah, no, the list comprehension just made a new list and assigned it to that variable name.

![image](https://user-images.githubusercontent.com/19836026/100628079-1b9e4e00-3328-11eb-9ec5-2b8054b99d7a.png)